src/PrestaShopBundle/Controller/Admin/CategoryController.php

```diff
@@ -92,12 +93,12 @@ public function addSimpleCategoryFormAction(Request $request)
                             ],
                         ]
                     );
-
                     if ($request->query->has('id_product')) {
-                        $productAdapter = $this->get('prestashop.adapter.data_provider.product');
-                        $product = $productAdapter->getProduct($request->query->get('id_product'));
-                        $product->addToCategories($categoryId->getValue());
-                        $product->save();
+                        $addCategoryToProductCommand = new AddCategoryToProductCommand(
```

Comment:
`AddCategoryToProductCommand` throws `CannotAddCategoryToProductException` that extends `ProductException` so the catch below will not be able to catch it.

Can you also catch this Exception and provide a nice displayable error message for it ? See for example
https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php#L154

Reply:
Done, I now return the correct error message, returned in the json.